client/lib/deploy.rs

```diff
@@ -76,31 +76,37 @@ impl From<GetBlockResult> for ListDeploysResult {
 pub(super) enum OutputKind<'a> {
     File {
         path: &'a str,
+        tmp_path: String,
         overwrite_if_exists: bool,
     },
     Stdout,
 }
 
 impl<'a> OutputKind<'a> {
     pub(super) fn file(path: &'a str, overwrite_if_exists: bool) -> Self {
+        let tmp_path = [path.clone(), ".tmp"].join("");
```

Comment:
Probably more canonical to do:

```suggestion
        let tmp_path = Path::new(path).with_extension(".tmp");
```